rest/route/host.go

```diff
@@ -312,3 +313,77 @@ func getLimit(vals url.Values) (int, error) {
 
 	return limit, nil
 }
+
+////////////////////////////////////////////////////////////////////////
+//
+// GET /host/range
+
+func makeFetchHostRange(sc data.Connector) gimlet.RouteHandler {
+	return &hostRangeGetHandler{
+		sc: sc,
+	}
+}
+
+type hostRangeGetHandler struct {
+	createdBefore time.Time
+	createdAfter  time.Time
+	distro        string
+	userSpawned   bool
+	status        string
+	mine          bool
+
+	sc data.Connector
+}
+
+func (h *hostRangeGetHandler) Factory() gimlet.RouteHandler {
+	return &hostRangeGetHandler{
+		sc: h.sc,
+	}
+}
+
+func (h *hostRangeGetHandler) Parse(ctx context.Context, r *http.Request) error {
+	body := util.NewRequestReader(r)
+	defer body.Close()
+	data := &model.APIHostParams{}
+	if err := util.ReadJSONInto(body, data); err != nil {
+		return errors.Wrap(err, "Argument read error")
+	}
+	h.createdBefore = data.CreatedBefore
+	h.createdAfter = data.CreatedAfter
+	h.distro = data.Distro
+	h.status = data.Status
+	h.userSpawned = data.UserSpawned
+	h.mine = data.Mine
```

Comment:
why not just have a field be mode.APIHostParams{} rather than copying all the values here?

Reply:
done